andela/ah-kgl-avengers-backend
# #165020209 Add endpoint for getting all ratings submitted for an article

## What does this PR do?
Add endpoint for getting all user ratings on an article

## Description of Task to be completed?
- Store user ID when a user rates an article
- Limit the number of times a user can rate an article to one
- Return all ratings and the profile of users who rated the article

## How should this be manually tested?
Using Postman or Insomnia:
1. Send a **POST** request to `/articles/:slug/ratings` replacing slug with the slug of the article to rate. The request body should be a `rating=number >0 & <6` The response should contain the article and the average rating that the article has.
2. Send a **GET** request to `/articles/:slug/ratings`, the response should be an array of objects having the rating and the profile of the user who submitted the rating.

## Any background context you want to provide?
You might need to add the `SERVER_ADDRESS=localhost:3000` field in your .env file for email verification to work.

## What are the relevant pivotal tracker stories?
[Users should be able to fetch article's ratings #165020209](https://www.pivotaltracker.com/n/projects/2321840)

## Screenshots
<img width="683" alt="Screen Shot 2019-04-30 at 07 21 41" src="https://user-images.githubusercontent.com/24508902/56942083-dba63200-6b18-11e9-83c7-c675efa6c37b.png">

<img width="794" alt="Screen Shot 2019-04-30 at 07 21 54" src="https://user-images.githubusercontent.com/24508902/56942095-ee206b80-6b18-11e9-8689-44e47e9a3a08.png">
